### ISSUE TITLE
Exception: The request signature we calculated does not match the signature you provided. Check your key and signing method.

### ISSUE THREAD

**Issue by superhare2014** [minio/minio-dotnet#386]
The Code :
```
// Server side copyObject with metadata replacement
Cases.CopyObjectMetadata.Run(minioClient, bucketName, objectName, destBucketName, objectName).Wait();
```
In Minio.Examples project will reproduce this issues.

**Comment by superhare2014** [comment]
This is client log:

`- - - - - - - - - - BEGIN REQUEST - - - - - - - - - -

GET https://play.min.io/minio-dotnet-example-h6y1h?location HTTP/1.1
Host: play.min.io:443
x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
x-amz-date: 20200330T142915Z
Authorization: AWS4-HMAC-SHA256 Credential=Q3AM3UQ867SPQQA43P2F/20200330/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=36bc3bda59433ca710cf61fbbc6845193e886605a9253ff158b4f4036a74db5e
Accept: application/json, application/xml, text/json, text/x-json, text/javascript, text/xml


- - - - - - - - - - END REQUEST - - - - - - - - - -

- - - - - - - - - - BEGIN RESPONSE - - - - - - - - - -

HTTP/1.1 404 NotFound
Server: nginx/1.10.3, (Ubuntu)
Date: Mon, 30 Mar 2020 14:29:19 GMT
Connection: keep-alive
Accept-Ranges: bytes
Content-Security-Policy: block-all-mixed-content
Vary: Origin
X-Amz-Request-Id: 16011B4C77A4EF8F
X-Xss-Protection: 1; mode=block
Content-Type: application/xml
Content-Length: 324


<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>NoSuchBucket</Code><Message>The specified bucket does not exist</Message><BucketName>minio-dotnet-example-h6y1h</BucketName><Resource>/minio-dotnet-example-h6y1h</Resource><RequestId>16011B4C77A4EF8F</RequestId><HostId>96fa3866-7ee6-4546-87d9-4283e3def6c3</HostId></Error>

- - - - - - - - - - END RESPONSE - - - - - - - - - -

[Bucket]  Exception: minio-dotnet-example-h6y1h: Not found.: Minio.Exceptions.BucketNotFoundException: MinIO API responded with message=Not found.
   at Minio.MinioClient.ParseError(IRestResponse response) in D:\Dev\TYArchiveManager\Source\test1\S3\minio-dotnet-master\Minio\MinioClient.cs:line 484
   at Minio.MinioClient.HandleIfErrorResponse(IRestResponse response, IEnumerable`1 handlers, DateTime startTime) in D:\Dev\TYArchiveManager\Source\test1\S3\minio-dotnet-master\Minio\MinioClient.cs:line 537
   at Minio.MinioClient.ExecuteTaskAsync(IEnumerable`1 errorHandlers, IRestRequest request, CancellationToken cancellationToken) in D:\Dev\TYArchiveManager\Source\test1\S3\minio-dotnet-master\Minio\MinioClient.cs:line 381
   at Minio.BucketRegionCache.Update(MinioClient client, String bucketName) in D:\Dev\TYArchiveManager\Source\test1\S3\minio-dotnet-master\Minio\BucketRegionCache.cs:line 104
   at Minio.MinioClient.GetRegion(String bucketName) in D:\Dev\TYArchiveManager\Source\test1\S3\minio-dotnet-master\Minio\MinioClient.cs:line 128
   at Minio.MinioClient.CreateRequest(Method method, String bucketName, String objectName, Dictionary`2 headerMap, String contentType, Object body, String resourcePath) in D:\Dev\TYArchiveManager\Source\test1\S3\minio-dotnet-master\Minio\MinioClient.cs:line 161
   at Minio.MinioClient.StatObjectAsync(String bucketName, String objectName, ServerSideEncryption sse, CancellationToken cancellationToken) in D:\Dev\TYArchiveManager\Source\test1\S3\minio-dotnet-master\Minio\ApiEndpoints\ObjectOperations.cs:line 827
   at Minio.MinioClient.CopyObjectAsync(String bucketName, String objectName, String destBucketName, String destObjectName, CopyConditions copyConditions, Dictionary`2 metadata, ServerSideEncryption sseSrc, ServerSideEncryption sseDest, CancellationToken cancellationToken) in D:\Dev\TYArchiveManager\Source\test1\S3\minio-dotnet-master\Minio\ApiEndpoints\ObjectOperations.cs:line 950
   at Minio.Examples.Cases.CopyObjectMetadata.Run(MinioClient minio, String fromBucketName, String fromObjectName, String destBucketName, String destObjectName) in D:\Dev\TYArchiveManager\Source\test1\S3\minio-dotnet-master\Minio.Examples\Cases\CopyObjectMetadata.cs:line 48`

**Comment by superhare2014** [comment]
this is client log:
`- - - - - - - - - - BEGIN REQUEST - - - - - - - - - -

PUT https://play.min.io/minio-dotnet-example-ymtxr/minio-dotnet-example-zb1n6 HTTP/1.1
Host: play.min.io:443
Content-Type: application/css
x-amz-meta-mynewkey: my-new-value
x-amz-copy-source: minio-dotnet-example-8kwqc/minio-dotnet-example-zb1n6
x-amz-metadata-directive: REPLACE
x-amz-content-sha256: UNSIGNED-PAYLOAD
x-amz-date: 20200330T143301Z
Authorization: AWS4-HMAC-SHA256 Credential=Q3AM3UQ867SPQQA43P2F/20200330/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-copy-source;x-amz-date;x-amz-metadata-directive;x-amz-meta-mynewkey, Signature=959c5ca776467eb24752510e9c06943fe720973d8fd14adb89ffab168207836a
Accept: application/json, application/xml, text/json, text/x-json, text/javascript, text/xml


- - - - - - - - - - END REQUEST - - - - - - - - - -

- - - - - - - - - - BEGIN RESPONSE - - - - - - - - - -

HTTP/1.1 403 Forbidden
Server: nginx/1.10.3, (Ubuntu)
Date: Mon, 30 Mar 2020 14:33:09 GMT
Connection: keep-alive
Accept-Ranges: bytes
Content-Security-Policy: block-all-mixed-content
Vary: Origin
X-Amz-Request-Id: 16011B82236516E3
X-Xss-Protection: 1; mode=block
Content-Type: application/xml
Content-Length: 475


<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>SignatureDoesNotMatch</Code><Message>The request signature we calculated does not match the signature you provided. Check your key and signing method.</Message><Key>minio-dotnet-example-zb1n6</Key><BucketName>minio-dotnet-example-ymtxr</BucketName><Resource>/minio-dotnet-example-ymtxr/minio-dotnet-example-zb1n6</Resource><RequestId>16011B82236516E3</RequestId><HostId>96fa3866-7ee6-4546-87d9-4283e3def6c3</HostId></Error>

- - - - - - - - - - END RESPONSE - - - - - - - - - -

[Bucket]  Exception: The request signature we calculated does not match the signature you provided. Check your key and signing method.: Minio.Exceptions.MinioException: MinIO API responded with message=The request signature we calculated does not match the signature you provided. Check your key and signing method.
   at Minio.MinioClient.ParseError(IRestResponse response) in D:\Dev\TYArchiveManager\Source\test1\S3\minio-dotnet-master\Minio\MinioClient.cs:line 392
   at Minio.MinioClient.HandleIfErrorResponse(IRestResponse response, IEnumerable`1 handlers, DateTime startTime) in D:\Dev\TYArchiveManager\Source\test1\S3\minio-dotnet-master\Minio\MinioClient.cs:line 537
   at Minio.MinioClient.ExecuteTaskAsync(IEnumerable`1 errorHandlers, IRestRequest request, CancellationToken cancellationToken) in D:\Dev\TYArchiveManager\Source\test1\S3\minio-dotnet-master\Minio\MinioClient.cs:line 381
   at Minio.MinioClient.CopyObjectRequestAsync(String bucketName, String objectName, String destBucketName, String destObjectName, CopyConditions copyConditions, Dictionary`2 customHeaders, String resource, CancellationToken cancellationToken, Type type) in D:\Dev\TYArchiveManager\Source\test1\S3\minio-dotnet-master\Minio\ApiEndpoints\ObjectOperations.cs:line 1044
   at Minio.MinioClient.CopyObjectAsync(String bucketName, String objectName, String destBucketName, String destObjectName, CopyConditions copyConditions, Dictionary`2 metadata, ServerSideEncryption sseSrc, ServerSideEncryption sseDest, CancellationToken cancellationToken) in D:\Dev\TYArchiveManager\Source\test1\S3\minio-dotnet-master\Minio\ApiEndpoints\ObjectOperations.cs:line 999
   at Minio.Examples.Cases.CopyObjectMetadata.Run(MinioClient minio, String fromBucketName, String fromObjectName, String destBucketName, String destObjectName) in D:\Dev\TYArchiveManager\Source\test1\S3\minio-dotnet-master\Minio.Examples\Cases\CopyObjectMetadata.cs:line 48`
